packages/ai/src/ai/modules/task/task_engine.py

```
@@ -313,10 +321,12 @@ def _resolve_pipeline(self, pipeline: Dict[str, Any]) -> Dict[str, Any]:
         # Convert dict to JSON string
         pipeline_str = json.dumps(pipeline)
 
-        # Replace ${VAR_NAME} with environment variable value
+        # Replace ${VAR_NAME} with environment variable value (if allowed)
         def replacer(match):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `replacer`

(ANN202)